src/HttpRequest.cc

```diff
@@ -773,11 +773,19 @@ UpdateRequestNotes(ConnStateData *csd, HttpRequest &request, NotePairs const &he
 void
 HttpRequest::manager(const CbcPointer<ConnStateData> &aMgr, const AccessLogEntryPointer &al)
 {
+    if (clientConnectionManager == aMgr)
+        return;
+
     clientConnectionManager = aMgr;
 
     if (!clientConnectionManager.valid())
         return;
 
+    // TODO: We fill request notes here until we find a way to verify whether
+    // no ACL checking is performed before ClientHttpRequest::doCallouts().
```

Comment:
I copied this TODO from buildHttpRequest(), for now.  I'm not sure whether it is relevant now, there are some ACL checking before ClientHttpRequest::doCallouts().